auth/gcloud/aio/auth/token.py

```diff
@@ -209,27 +222,72 @@ async def ensure_token(self) -> None:
             self.acquire_access_token())
         await self.acquiring
 
-    async def _refresh_authorized_user(self, timeout: int) -> Response:
+    @abstractmethod
+    async def refresh(self, *, timeout: int) -> TokenResponse:
+        pass
```

Comment:
Same as https://docs.python.org/3/library/exceptions.html#NotImplementedError.

Reply:
Ditto for my comment above.
https://github.com/talkiq/gcloud-aio/pull/637#discussion_r1333407708